lib/engine/game/g_18_pa/game.rb

```diff
@@ -145,23 +143,89 @@ class Game < Game::Base
+            Engine::Step::SelectionAuction,
+          ])
+        end
+
+        def stock_round
+          Engine::Round::Stock.new(self, [
+            Engine::Step::DiscardTrain,
+            Engine::Step::Exchange,
+            Engine::Step::SpecialTrack,
+            Engine::Step::BuySellParShares,
+          ])
+        end
+
         def operating_round(round_num)
           Round::Operating.new(self, [
             Engine::Step::Bankrupt,
-            Engine::Step::Exchange,
-            Engine::Step::SpecialTrack,
             Engine::Step::SpecialToken,
-            Engine::Step::BuyCompany,
-            Engine::Step::HomeToken,
-            Engine::Step::Track,
+            G18PA::Step::Track,
             Engine::Step::Token,
             Engine::Step::Route,
-            Engine::Step::Dividend,
+            G18PA::Step::Dividend,
             Engine::Step::DiscardTrain,
             Engine::Step::BuyTrain,
-            [Engine::Step::BuyCompany, { blocks: true }],
           ], round_num: round_num)
         end
+
+        def setup
+          @scranton_marker_ability = Engine::Ability::Description.new(type: 'description', description: 'Scranton Token')
+
+          # place the home station for all corporations and minors except NYC.
+          @corporations.each do |corporation|
+            next if corporation.id == 'NYC'
+
+            tile = hex_by_id(corporation.coordinates)&.tile
+            tile.cities[corporation.city || 0].place_token(corporation, corporation.tokens.first, free: true)
+          end
+        end
+
+        def scranton_marker_available?
+          hex_by_id(SCRANTON_HEX).tile.icons.any? { |icon| icon.name == SCRANTON_MARKER_ICON }
+        end
+
+        def scranton_marker?(entity)
+          return false if !entity.corporation? || entity.type == :minor
+
+          scranton_markers(entity).any?
+        end
+
+        def scranton_markers(entity)
+          entity.all_abilities.select { |ability| ability.description == @scranton_marker_ability.description }
+        end
+
+        def connected_to_scranton?(entity)
+          graph.reachable_hexes(entity).include?(hex_by_id(SCRANTON_HEX))
+        end
+
+        def can_buy_scranton_marker?(entity)
+          return false if !entity.corporation? || entity.type == :minor
+
+          scranton_marker_available? &&
+            !scranton_marker?(entity) &&
+            buying_power(entity) >= SCRANTON_MARKER_COST &&
+            connected_to_scranton?(entity)
+        end
+
+        def buy_scranton_marker(entity)
+          return unless can_buy_scranton_marker?(entity)
+
+          entity.spend(SCRANTON_MARKER_COST, @bank)
+          entity.add_ability(@scranton_marker_ability.dup)
+          @log << "#{entity.name} buys a Scranton bonus token for $#{SCRANTON_MARKER_COST}."
+
+          tile_icons = hex_by_id(SCRANTON_HEX).tile.icons
+          tile_icons.delete_at(tile_icons.find_index { |icon| icon.name == SCRANTON_MARKER_ICON })
```

Comment:
The codebase style is to prefer to use `Array.index` instead of `Array.find_index`.
```suggestion
          tile_icons.delete_at(tile_icons.index { |icon| icon.name == SCRANTON_MARKER_ICON })
```

Reply:
fixed